### PR TITLE
Use args passed to CreateBuilderAsync in GetCurrentExecutableInfo

### LINKED ISSUE #5239: [MTP] `GetCurrentExecutableInfo` shouldn't calculate command line arguments via `Environment`
Whatever was passed to `CreateBuilderAsync` should be used instead.

For example, if I write my custom `Main` and do the following:

```
if (args.Length == 0)
{
    args = ["--retry-failed-tests", "1"];
}

var builder = await TestApplication.CreateBuilderAsync(args);
SelfRegisteredExtensions.AddSelfRegisteredExtensions(builder, args);
using var app = await builder.BuildAsync();
return await app.RunAsync();
```

Retry extension will throw an exception, because `GetCurrentExecutableInfo` doesn't have the correct arguments.

### PR DESCRIPTION
**Pull request by Evangelink**
Fixes #5239.

Previously, `CurrentTestApplicationModuleInfo.GetCurrentExecutableInfo` derived its arguments from `Environment.GetCommandLineArgs()`. If a custom `Main` mutated the args before forwarding them to `TestApplication.CreateBuilderAsync` (for example, to inject `--retry-failed-tests 1` as a default), the platform options reflected the mutation but the executable info did not. Extensions like Retry that re-launch the test host using these arguments would then throw.

### Fix

`CurrentTestApplicationModuleInfo` now accepts the args passed to `CreateBuilderAsync` and uses them when computing `GetCurrentExecutableInfo`. The DLL path is still recovered from `Environment.GetCommandLineArgs()[0]` for the `dotnet`/`mono` muxer cases, preserving the existing behavior when running under `dotnet.exe`.

### Repro from the issue

```csharp
if (args.Length == 0)
{
    args = ["--retry-failed-tests", "1"];
}

var builder = await TestApplication.CreateBuilderAsync(args);
SelfRegisteredExtensions.AddSelfRegisteredExtensions(builder, args);
using var app = await builder.BuildAsync();
return await app.RunAsync();
```

With the fix, the Retry extension now sees `[--retry-failed-tests, 1]` in the executable info regardless of what was on the OS command line.

### Tests

Added `CurrentTestApplicationModuleInfoTests` covering the AppHost and dotnet muxer paths with and without user-passed args (5 new tests). 290 related unit tests continue to pass.

### CODE REVIEW

File: src/Platform/Microsoft.Testing.Platform/Services/CurrentTestApplicationModuleInfo.cs [hunk 1]
```diff
@@ -5,12 +5,25 @@
...
+    public CurrentTestApplicationModuleInfo(IEnvironment environment, IProcessHandler process)
+        : this(environment, process, commandLineArguments: null)
+    {
+    }
+
+    public CurrentTestApplicationModuleInfo(IEnvironment environment, IProcessHandler process, string[]? commandLineArguments)
+    {
+        _environment = environment;
+        _process = process;
+        _commandLineArguments = commandLineArguments;
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply src/Platform/Microsoft.Testing.Platform/Services/CurrentTestApplicationModuleInfo.cs]: Thanks - handled in 286d71d. The constructor now takes a defensive copy via `(string[])commandLineArguments.Clone()`, and `GetCurrentExecutableInfo_PassedArgsAreSnapshotted_CallerMutationDoesNotAffectResult` covers the snapshot behavior.

File: src/Platform/Microsoft.Testing.Platform/Services/CurrentTestApplicationModuleInfo.cs [hunk 2]
```diff
@@ -102,16 +115,49 @@ public ExecutableInfo GetCurrentExecutableInfo()
...
+                    ? [.. MuxerExec, envArgs[0], .. passedArgs]
+                    : [.. MuxerExec, .. passedArgs];
+            }
+            else
+            {
+                // Mono muxer: prepend the dll path (envArgs[0]) before user-supplied args.
+                string[] envArgs = _environment.GetCommandLineArgs();
+                arguments = envArgs.Length > 0
+                    ? [envArgs[0], .. passedArgs]
+                    : passedArgs;
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply src/Platform/Microsoft.Testing.Platform/Services/CurrentTestApplicationModuleInfo.cs:148]: Thanks - handled in 286d71d. Added three mono muxer tests (no passed args, with passed args, and empty `envArgs` edge case) plus the matching dotnet-muxer empty `envArgs` case.
